src/vec.rs

```diff
@@ -247,26 +247,40 @@ macro_rules! impl_vecn_float_methods {
         /// Returns `self` normalized to length 1.0.
         ///
         /// For valid results, `self` must _not_ be of length zero, nor very close to zero.
-        /// You can check the results with [`Self::is_finite`], or use [`Self::normalize_or_zero`].
+        ///
+        /// See also [`Self::try_normalize`] and [`Self::normalize_or_zero`].
         #[inline(always)]
         pub fn normalize(self) -> Self {
             Self($flttrait::normalize(self.0))
         }
 
-        /// Returns `self` normalized to length 1.0 if possible, else returns zero.
+        /// Returns `self` normalized to length 1.0 if possible, else returns None.
         ///
         /// In particular, if the input is zero (or very close to zero), or non-finite,
-        /// the result of this operation will be zero.
-        #[inline(always)]
-        pub fn normalize_or_zero(self) -> Self {
+        /// the result of this operation will be None.
+        ///
+        /// See also [`Self::normalize_or_zero`].
+        #[inline]
+        pub fn try_normalize(self) -> Option<Self> {
             let rcp = self.length_recip();
             if rcp.is_finite() && rcp > 0.0 {
-                self * rcp
+                Some(self * rcp)
             } else {
-                Self::ZERO
+                None
             }
         }
 
+        /// Returns `self` normalized to length 1.0 if possible, else returns zero.
+        ///
+        /// In particular, if the input is zero (or very close to zero), or non-finite,
+        /// the result of this operation will be zero.
+        ///
+        /// See also [`Self::try_normalize`].
+        #[inline]
+        pub fn normalize_or_zero(self) -> Self {
+            self.try_normalize().unwrap_or(Self::ZERO)
```

Comment:
Using `try_normalize` here adds overhead which isn't really necessary, we could live with some code duplication to avoid using `Option` here.